### DISCUSSION TITLE
How to use @itwin/editor-frontend to move the iModel

### DISCUSSION THREAD

**Discussion by salah-walid** [iTwin/itwinjs-core#5939] [Q&A] [answered]
Using the `@itwin/editor-frontend` how to move the iModel in 3D space like the example shown here:
![image](https://github.com/iTwin/itwinjs-core/assets/29980658/dbae75ea-f403-4905-b46f-7f336cffa9c7)
https://www.npmjs.com/package/@itwin/editor-frontend?activeTab=readme

I made sure to initialize the functionalities
```TypeScript
    await EditTools.initialize()
```

Then launched the tool like this:
```TypeScript
    IModelApp.tools.run(MoveElementsTool.toolId)
```

**Comment by bbastings** [comment]
You are trying to geolocate the imodel and you've opened it for write?

The key-in to display the project location handles shown in that image is `editor project location show`. The tool that runs is ProjectLocationShowTool.

Then select handle #2 (circle w/X) and snap to a known location, you will be prompted to enter the latitude, longitude, and altitude of that location in the tool settings dialog. You can also select handle #3 (north arrow) to change the direction if needed.

Lastly, write the changes using `editor project location save`. The tool that runs is ProjectLocationSaveTool.

MoveElementsTool is for selecting a set elements and applying a translation to their placement, so not applicable to geolocation.

**Reply by salah-walid** [reply]
How to input those key-ins ?

**Comment by bbastings** [comment, marked as answer]
You can run the tools that were mentioned above (also in the Readme) just like you did the MoveElementsTool or you can make the key-in ui available in your application (Ctrl+F2 might bring up the key-in palette by default?).

**Reply by salah-walid** [reply]
It seems [from this](https://github.com/iTwin/itwinjs-core/blob/6c9bd38e3ee8b465eb09fd8c8874a21a38b9c970/editor/frontend/src/ProjectLocation/ProjectExtentsDecoration.ts#L960) that the imodel has to be setup with a briefcase.
My question now how to do that, i suppose inside the frontstage

**Reply by aruniverse** [reply]
Are you trying to modify the extents within a web or desktop application?
If you are using the web viewer, it will not use a briefcase connection. You will need to use a desktop app that has a writeable briefcase connection.
